src/auth/middleware/bearer.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const { users } = require('../../models/index');
+
+module.exports = async (req, res, next) => {
+
+  try {
+
+    if (!req.headers.authorization) { _authError(); }
+
+    const token = req.headers.authorization.split(' ').pop();
+    const validUser = await users.authenticateToken(token);
+    req.user = validUser;
+    req.token = validUser.token;
+    next();
+
+  } catch (e) {
+    _authError();
+  }
```

Comment:
If it throws an error, the bearer function catches it and runs the `_authError()` helper function.